src/index_runner/elasticsearch_consumer.py

```diff
@@ -21,10 +34,63 @@ def main():
         threadify(_save_to_elastic, [msg_data])
 
 
+def _validate_message(msg_data):
+    """
+    validate the input message
+    """
+    if not msg_data.get('doc'):
+        raise RuntimeError("Message to Elasticsearch malformed, does not contain \'doc\'")
+    if not msg_data.get('id'):
+        raise RuntimeError("Message to Elasticsearch malformed, does not contain \'id\' field")
+    if not msg_data.get('index'):
+        raise RuntimeError("Message to Elasticsearch malformed, does not have target elasticsearch index")
+
+
 def _save_to_elastic(msg_data):
     """
     Save the indexed doc to elasticsearch.
+    msg_data:
+        {
+            'doc': elasticsearch index document
+                   - json like object
+            'mapping': elasticsearch type mapping schema thing
+        }
     """
+    try:
+        _validate_message(msg_data)
+    except RuntimeError as error:
+        # log the error
+        msg_data['error'] = str(error)
+        producer.produce(
+            config['topics']['error_logs'],
+            json.dumps(msg_data),
+            callback=_delivery_report
+        )
+        producer.poll(60)
+        raise error
+
+    try:
+        # save to elasticsearch index
+        resp = requests.put(
+            '/'.join([es_url, msg_data['index'], es_data_type, msg_data['id']]),
+            data=json.dumps(msg_data['doc']),
+            headers=headers
+        )
+    except requests.exceptions.RequestException as error:
+        # log the error
+        msg_data['error'] = str(error)
+        producer.produce(
+            config['topics']['error_logs'],
+            json.dumps(msg_data),
+            callback=_delivery_report
+        )
+        producer.poll(60)
+        raise error
+    if not (resp.status_code == requests.codes.ok):
```

Comment:
I think that using `resp.ok` will cover all 2xx responses. `requests.codes.ok` doesn't work well sometimes because it only covers a 200 response, not 201, 203, etc which are all successful. So instead you could do `if not resp.ok: ...`